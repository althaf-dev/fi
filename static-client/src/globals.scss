@import '../src/styles/font';
@import '../src/styles/base';
@import '../src/styles/keyframes';
@import '../src/styles/default';
@import '../src/styles/transition';

* {
  box-sizing: border-box;
  padding: 0;
  margin: 0;
}

html,
body {
  max-width: 100vw;
  min-height: 100vh;
  overflow-x: hidden;
}

body {
  color: rgb(var(--foreground-rgb));
}

a {
  color: inherit;
  text-decoration: none;
}


@media only screen and (min-width: 360px) {
  .blog {
    max-width: 320px;
    margin: 0 auto;
  }
  .blog h2,h1 {
    font-size: 20px;
    line-height: 24px;
    padding-bottom: 14px;
    font-weight: 600;
    color: #313234;
  }

  .blog h3{
    font-size: 16px;
    line-height: 24px;
    padding-bottom: 14px;
    font-weight: 600;
    color: #313234;
  }

  .blog p {
    font-size: 16px;
    line-height: 24px;
    padding-bottom: 32px;
    color: #313234 !important;
    font-weight: 400;
    font-family: Inter;
  }

  .blog ul {
    margin-left: 15px;
    font-family: Gilroy;
    font-weight: 600;
    font-size: 20px;
    line-height: 34px;
    color:#6A6D70;
  }

  .blog ul li , .blog ol li{
    font-family: Inter;
    font-weight: 400;
    font-size: 16px;
    line-height: 24px;
    margin-left: 20px;
  }

  
  .blog ul li span span strong, .blog ol li span span strong{
    font-size: 20px  !important;
    line-height: 34px !important;
    padding: 0 0 12px 0;
    font: Gilroy !important;
    font-weight: 600 !important;
    color:#6A6D70;
  }

  .blog ul li, .blog ol li {
    padding: 0 0 24px 0;
  }

  .blog blockquote p:nth-of-type(1) {
    font-weight: 400;
    font-size: 20px;
    line-height: 32px;
    color: #929599; 
    font-family: Inter;
  }


  .blog blockquote {
    border-left: 4px solid #6BCDB6;
    padding: 0 0 0 24px;
    font-weight: 400;
    font-size: 20px;
    line-height: 32px;
    color: #929599; 
    font-family: Inter;
    margin: 12px 0;
  }

  .blog .banner {
    width: 300px !important;
    border-radius: 22px;
    padding: 24px;
    display: flex;
    gap: 25px;
    margin: 32px 0;
  }

  .blog .banner-image {
    width: 92px !important;
    height: 113px !important;
  }

  .blog .banner-title {
    font-family: Inter;
    font-weight: 700;
    font-size: 16px;
    line-height: 20px;
    margin-bottom: 16px;
  }

  .blog .banner-sub-title {
    font-family: Inter;
    font-weight: 700;
    font-size: 16px;
    line-height: 20px;
    display: flex;
    gap: 5px;
    align-items: center;
  }

  .blog .banner-arrow {
    width: 18px !important;
    height: 18px !important;
    margin-bottom: 0px !important;
  }

  .blog .banner-link {
    display: flex;
    gap: 5px;
    align-items: center;
  }

  .blog h4 {
    font-family: Gilroy;
    font-weight: 700;
    font-size: 20px;
    line-height: 24px;
    color: #313234;
    margin-bottom: 20px;
  }

  .blog .table {
    width: 300px;
    margin-bottom: 32px;
    border-radius: 24px;
    overflow-x: scroll;
  }

  .blog table {
    border-collapse: collapse;
    width: 300px;
    background-color: white;
    margin-bottom: 12px;
  }

  .blog table tbody tr:nth-of-type(1) {
    font-family: Gilroy;
    font-weight: 600;
    font-size: 16px;
    line-height: 20px;
    color: #57595D !important;
  }

  .blog table tbody tr {
    font-family: Gilroy;
    font-weight: 600;
    font-size: 16px;
    line-height: 20px;
    margin-bottom: 14px;
  }

  .blog table tbody tr td {
    border: 1px solid #ddd;
    padding: 28px;
  }

  .blog table {
    border-collapse: separate;
    border-spacing: 0;
  }

  .blog td {
    border-style: none solid solid none;
    padding: 10px;
  }

  .blog tr:first-child td:first-child {
    border-top-left-radius: 24px;
  }

  .blog tr:first-child td:last-child {
    border-top-right-radius: 24px;
  }

  .blog tr:last-child td:first-child {
    border-bottom-left-radius: 24px;
  }

  .blog tr:last-child td:last-child {
    border-bottom-right-radius: 24px;
  }

  .blog tr:first-child td {
    border-top-style: solid;
  }

  .blog tr td:first-child {
    border-left-style: solid;
  }
  
  .blog img{
    width: 100% !important;
    height: 10rem !important;
    border-radius: 20px;
    margin-bottom: 20px;
  }

  .blog .image-content-desktop{
    display: none;
  }

  .blog .image-content-mobile{
    display: inline-block;
    max-width: 356px;
    height: auto;
    border-radius: 20px;
    object-fit: contain;
  }

  .video iframe{
    width: 80% !important;
    height: 10rem !important;
    border-radius: 20px;
    display: flex;
    margin: 20px auto;
  }

  .blog a{
    color: #00B899;
  }
}

@media only screen and (min-width: 768px) {

  .blog {
    max-width: 550px;
  }

  .blog .table {
    width: 550px;
  }

  .blog table {
    width: 550px;
  }

  .blog .image-content-desktop{
    display: inline-block;
    max-width: 827px;
    border-radius: 24px;
    margin-bottom: 24px;
    width: 100%;
    height: auto;
    object-fit: contain;
  }

  .blog .image-content-mobile{
    display: none;
  }

  .blog .banner {
    width: 550px !important;
  }

  .video iframe{
    width: 80% !important;
    height: 15rem !important;
    border-radius: 20px;
    margin: 20px auto;
  }

}

@media only screen and (min-width: 1440px){
  .blog{
    max-width: 800px;
    
    h2, h1{
      font: Gilroy;
      font-weight: 600;
      font-size: 28px;
      line-height: 32px;
      padding-bottom: 24px;
    }

    h3{
      font: Gilroy;
      font-weight: 600;
      font-size: 24px;
      line-height: 32px;
      padding-bottom: 24px;
    }
    
    p{
      font-family: Inter;
      font-weight: 400;
      font-size: 24px;
      line-height: 36px;
      color: #313234 !important;
      padding-bottom: 40px;
    }
    
    img{
      width: 100% !important;
      height: 25rem !important;
      border-radius: 24px;
      margin-bottom: 24px;
    }

    .video iframe{
      width: 80% !important;
      height: 25rem !important;
      border-radius: 24px;
      margin: 50px auto;
    }
  
    ul, ol{
      margin-left: 30px;
      color:#6A6D70;
    }
  
    ul li span span strong,   ol li span span strong{
      font: Gilroy !important;
      font-weight: 600 !important;
      font-size: 28px  !important;
      line-height: 32px !important;
      color:#6A6D70;
      padding: 0 0 16px 0;
    }
  
    ul, ol {
      font: Gilroy;
      font-weight: 600;
      font-size: 28px;
      line-height: 32px;
    }
  
    ul li,   ol li{
      font: Inter;
      font-weight: 500;
      font-size: 24px;
      line-height: 36px;
      color:#313234;
    }
    ul li, ol li{
      padding:0 0 24px 0;
    }
  
    blockquote p:nth-of-type(1){
      font-family: Inter;
      font-weight: 400;
      font-size: 32px;
      line-height: 48px;
      color: #929599 !important; 
    }
  
    blockquote {
      border-left: 4px solid #6BCDB6;
      margin:0 0 0 24px;
      margin: 24px 0;
      font-family: Inter;
      font-weight: 400;
      font-size: 32px;
      line-height: 48px;
      color: #929599 !important; 
    }
  
    .banner{
      width: 826px !important;
      border-radius: 24px;
      padding: 40px 75px;
      display: flex;
      gap: 100px;
      margin: 32px 0;
    }
    
    .banner-image{
      width: 162px ! important;
      height: 199px ! important;
    }
    
    .banner-title{
      font: Gilroy;
      font-weight: 600;
      font-size: 32px;
      line-height: 38px;
      margin-bottom: 32px;
    }
    
    .banner-sub-title{
      font: Inter;
      font-weight: 700;
      font-size: 24px;
      line-height: 20px;
      text-decoration: underline;
    }
    
    .banner-arrow{
      width: 20px ! important;
      height: 20px ! important;
      margin-bottom: 0px !important;
    }
    
    .banner-link{
      display: flex;
      align-items: center;
      gap:9px;
    }
    h4{
      font: Gilroy;
      font-weight: 600;
      font-size: 20px;
      line-height: 24px;
      color:#313234;
      margin-bottom: 24px;
    
    }
    .table{
      width: 827px;
      margin-bottom: 32px;
      border-radius: 24px;
    }
    
    table{
      border-collapse: collapse;
      width: 827px;
      margin-bottom: 24px;
    }
    
    table tbody tr:nth-of-type(1){
      font: Gilroy;
      font-weight: 600;
      font-size: 20px;
      line-height: 24px;
    }
    
    table tbody tr{
      font: Gilroy;
      font-weight: 600;
      font-size: 20px;
      line-height: 24px;
      margin-bottom: 24px;
    }
    
    table tbody tr td{
      border: 1px solid #ddd;
      padding: 28px;
    }

    table { border-collapse: separate; border-spacing: 0; }
    
    td {
        border-style: none solid solid none;
        padding: 10px;
    }
    tr:first-child td:first-child { border-top-left-radius: 24px; }
    tr:first-child td:last-child { border-top-right-radius: 24px; }
    tr:last-child td:first-child { border-bottom-left-radius: 24px; }
    tr:last-child td:last-child { border-bottom-right-radius: 24px; }
    tr:first-child td { border-top-style: solid; }
    tr td:first-child { border-left-style: solid; }
  
  }

}
