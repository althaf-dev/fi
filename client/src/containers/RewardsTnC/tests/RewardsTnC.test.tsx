/* eslint-disable no-undef, import/no-extraneous-dependencies */
import React from 'react';
import { expect } from '@jest/globals';

import { render } from '../../../testing/test-utils';
import RewardsTnCHome from '../index';

it('should render RewardsTnC component', () => {
    const { asFragment } = render(<RewardsTnCHome />);
    expect(asFragment()).toMatchSnapshot();
});
