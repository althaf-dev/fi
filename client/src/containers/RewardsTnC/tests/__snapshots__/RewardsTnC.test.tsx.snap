// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render RewardsTnC component 1`] = `
<DocumentFragment>
  <div
    class="sc-ckMVTt dYOwIC"
  >
    <div
      class="sc-fbPSWO bwNDJV"
    >
      <div
        class="sc-fYHEnZ jWbBCn"
      >
        <div
          class="sc-cQIpJi hGfYjM"
        >
          <img
            alt="Fi"
            class="sc-gsnTZi sc-dkzDqf krIvpr hKVGsE"
            loading="eager"
            src="logo.svg"
          />
        </div>
      </div>
      <div
        class="sc-ktCSKO JsBAF sc-gYMRRK gIJoSD"
      >
        <div
          class="sc-bGWzfD cxgLXy"
        >
          <div
            class="sc-IIEeM dcNKjL"
          >
            <h1
              class="sc-papXJ fEfRcZ sc-gGnURB gzxXgY"
              color="WHITE"
            >
              Rewards
            </h1>
            <div
              class="sc-eCYdqJ jVNGkM sc-iuStju jPsvXh"
              color="SILVER_2"
            >
              The Company has a rewards policy for Users of the Platform. To participate in the rewards programme of the Company (and win rewards) You must agree to these reward specific terms and conditions in addition to the general Terms and Conditions of the Company.
            </div>
          </div>
          <div
            class="sc-fytwQQ cTtcGf"
          >
            <img
              alt="image"
              class="sc-gsnTZi sc-dkzDqf krIvpr eoDeNX"
              loading="lazy"
              src="rewards-poster.webp"
              width="auto"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="sc-gFGZVQ ebEOJr"
    >
      <div
        class="sc-bcSJjp sc-coCPJf jshUfF cnRoei"
      >
        <div
          class="sc-ldZUWu eyOHuY"
        >
          <div
            class="sc-eCYdqJ iidbqz sc-cBQgDv sc-bcnBk bxeyxi kSuXzS"
            color="SUVA_GREY"
          >
            Yes, we have a cool rewards programme. But, it comes with specific stipulations. Including our right to discontinue offers when we choose to, stopping any user from using our Rewards to engage in fraudulent actions, and so on. We urge you to keep on reading, as this policy applies to all the users of the Fi app or Platform. Here's a one-liner description:
            <br />
            <br />
            To participate (and win) in our rewards programme, you must agree to our reward-specific and general
            <a
              href="/T&C"
              target="_blank"
            >
              <span
                class="sc-hcJkSI buNKGU"
                color="FOREST_GREEN"
              >
                ¬†T&Cs.
              </span>
            </a>
          </div>
        </div>
        <div
          class="sc-hCDzWh bXfKDn"
        >
          <div
            class="sc-eCYdqJ rrltT sc-cBQgDv bxeyxi"
            color="MINE_SHAFT"
          >
            1. General Terms
          </div>
          <div
            class="sc-eCYdqJ fcSnPF sc-cBQgDv bxeyxi"
            color="FOREST_GREEN"
          >
            2. User and Epifi
          </div>
          <div
            class="sc-eCYdqJ fcSnPF sc-cBQgDv bxeyxi"
            color="FOREST_GREEN"
          >
            3. Eligibility
          </div>
        </div>
      </div>
      <div
        class="sc-eBxihg RLLwy"
      />
      <div>
        <div
          class="sc-bcSJjp jshUfF"
        >
          <div
            class="sc-hCDzWh bXfKDn"
          >
            <div
              class="sc-eCYdqJ foTXFJ sc-lbhYTY fkwvct"
            >
              General Terms üìñ
            </div>
          </div>
          <div
            class="sc-ldZUWu eyOHuY"
          >
            <div
              class="sc-eCYdqJ iidbqz sc-fpcwso hprLYW"
              color="SUVA_GREY"
            >
              You agree that your participation in our rewards programme establishes that you've understood and agreed to these reward-specific T&Cs. The reward-specific T&Cs form a binding legal agreement between you and the Company. Having said this, please note that this is a discretionary programme of the Company.
              <br />
              At our sole discretion, we reserve the right to disqualify any user who does not meet the offer requirements to discontinue or change any offer or reward programme at any time. Sometimes, for cause, including but not limited to: any misuse of the offer, fraud, suspicious transaction or activity, and under any legal requirement or applicable rules and regulations. We also reserve the right to discontinue or change any offer or reward programme at any time, at our sole discretion.
            </div>
          </div>
        </div>
        <div
          class="sc-eBxihg RLLwy"
        />
      </div>
      <div>
        <div
          class="sc-bcSJjp jshUfF"
        >
          <div
            class="sc-hCDzWh bXfKDn"
          >
            <div
              class="sc-eCYdqJ foTXFJ sc-lbhYTY fkwvct"
            >
              User and Epifi ü§ù
            </div>
          </div>
          <div
            class="sc-ldZUWu eyOHuY"
          >
            <div
              class="sc-eCYdqJ iidbqz sc-fpcwso hprLYW"
              color="SUVA_GREY"
            >
              User understands that the reward amount for their Smart Deposit is a discretionary sum ‚Äì which will be given to them by Epifi Technologies Private Limited (Epifi). The sum will be allocated by Epifi exclusively for depositing in their Smart Deposit account (in their name) with Epifi's partner bank.
              <br />
              <br />
              Accordingly, User hereby grants and confirms the existence of the right of lien and set-off for the amount received by the User from Epifi (as a reward) for deposit in their Smart Deposit account. At its sole discretion and without notice to the User, the Company may utilise the amount received by the User, as a reward, towards their Smart Deposit account with Epifi's partner bank.
            </div>
          </div>
        </div>
        <div
          class="sc-eBxihg RLLwy"
        />
      </div>
      <div>
        <div
          class="sc-bcSJjp jshUfF"
        >
          <div
            class="sc-hCDzWh bXfKDn"
          >
            <div
              class="sc-eCYdqJ foTXFJ sc-lbhYTY fkwvct"
            >
              Eligibility ‚úÖ
            </div>
          </div>
          <div
            class="sc-ldZUWu eyOHuY"
          >
            <div
              class="sc-eCYdqJ iidbqz sc-jUtGcO sc-eMWWsl cTbcVF fmiDZx"
              color="SUVA_GREY"
            >
              Your eligibility for the Company's rewards programme isn't based on your compliance with these reward-specific T&Cs alone. You must also meet the eligibility criteria for each offer.
              <br />
              <br />
              Each reward's details, and individual eligibility T&Cs, appear on the Offers page of the Platform. Offers provided to a particular user may vary from user-to-user. You agree to read the offer T&Cs applicable to your offer ‚Äì as provided in your Epifi app or Platform account carefully.
              <br />
              <br />
              Please note that our Platform users residing in the State of Tamil Nadu are not eligible to participate in specific rewards or offers, as per the applicable laws in Tamil Nadu. Thus, users residing in Tamil Nadu are requested not to participate in offers relating to cash backs, etc.
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
